components/feature/contextmenu/src/test/java/mozilla/components/feature/contextmenu/ContextMenuCandidateTest.kt

```diff
@@ -1029,6 +1033,52 @@ class ContextMenuCandidateTest {
 
         verify(context).startActivity(any())
     }
+
+    @Test
+    fun `GIVEN SessionState with null EngineSession WHEN isUrlSchemeAllowed is called THEN it returns true`() {
+        val sessionState = TabSessionState(
+            content = mock(),
+            engineState = EngineState(engineSession = null)
+        )
+
+        assertTrue(sessionState.isUrlSchemeAllowed("http://mozilla.org"))
+    }
+
+    @Test
+    fun `GIVEN SessionState with no blocked url schemes WHEN isUrlSchemeAllowed is called THEN it returns true`() {
+        val noBlockedUrlSchemesEngineSession = Mockito.mock(EngineSession::class.java)
+        doReturn(emptyList<String>()).`when`(noBlockedUrlSchemesEngineSession).getBlockedSchemes()
+        val sessionState = TabSessionState(
+            content = mock(),
+            engineState = EngineState(engineSession = noBlockedUrlSchemesEngineSession)
+        )
+
+        assertTrue(sessionState.isUrlSchemeAllowed("http://mozilla.org"))
+    }
+
+    @Test
+    fun `GIVEN SessionState with blocked url schemes WHEN isUrlSchemeAllowed is called THEN it returns false if the url has that scheme`() {
+        val engineSessionWithBlockedUrlScheme = Mockito.mock(EngineSession::class.java)
+        doReturn(listOf("http")).`when`(engineSessionWithBlockedUrlScheme).getBlockedSchemes()
+        val sessionState = TabSessionState(
+            content = mock(),
+            engineState = EngineState(engineSession = engineSessionWithBlockedUrlScheme)
+        )
+
+        assertFalse(sessionState.isUrlSchemeAllowed("http://mozilla.org"))
```

Comment:
Can we add a few more edge cases here e.g. `hTtPs://` (which I know will work because you used `normalizedScheme` but good to guard against) and maybe a url without a protocol `mozilla.org` or `/path`.